### PR TITLE
Nima Quickstart Archetype

### LINKED ISSUE #5942: Nima WebServer quickstart archetype
Add a Nima WebServer Quickstart archetype for Helidon 4.0 that uses the Nima WebServer APIs.

### PR DESCRIPTION
**Pull request by tvallin**
fix #5942

### CODE REVIEW

**Review by barchetta — COMMENTED** [review]
Gave it a try and it works! Good start. A few things:
1. README: it states Java17+. Should be Java19+. I assume that is in a common file
2. README: `java -jar` command needs `--enable-preview`. You can go ahead and add that in common as well since MP now needs it and it does not hurt SE (which might need it at some point)
3. README: it's missing instructions on how to exercise the application. Maybe because missing custom? which brings me to...

I see there is no "custom" choice for nima apptype. I wonder if we should go ahead and create one even if it is not very featureful at first -- like maybe just json provider? This is to implement our design pattern even if it is not fully fleshed out yet.

File: archetypes/helidon/src/main/archetype/nima/common/common-nima.xml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+-->
+<archetype-script xmlns="https://helidon.io/archetype/2.0"
+                  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                  xsi:schemaLocation="https://helidon.io/archetype/2.0 https://helidon.io/xsd/archetype-2.0.xsd">
+    <exec src="/common/common.xml"/>
+    <source src="/common/sources.xml"/>
+    <output>
+        <model>
+            <value key="mainClass">${package}.Main</value>
+            <value key="parent-artifactId">helidon-se</value>
```
Comment (barchetta) [inline archetypes/helidon/src/main/archetype/nima/common/common-nima.xml:27]: Change the parent to `helidon-nima`. You will need to do this after PR #6232 is merged.

**Comment by tvallin** [comment]
The README can be modified without the `custom` (see latest commit). I think we can create a new issue for `custom` and make the changes there based on what features will be provided by `custom`. This way, `quickstart` is ready for next release thanks to this PR, and `custom` can have more features available than `media` for instance.

**Comment by barchetta** [comment]
I have created issue #6243 to track second phase of work.

**Review by barchetta — APPROVED** [review]

**Comment by tvallin** [comment]
@romain-grecourt would you like to have a look at it before merging ?

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**